apollo/src/graphql/storage-provider/storageProviderQueries.ts

```diff
@@ -27,4 +30,96 @@ builder.queryFields((t) => ({
             return storageProvider;
         },
     }),
+    listObjectsForModelVersion: t.field({
+        type: StorageProviderObjectConnection,
+        authScopes: {
+            loggedIn: true,
+        },
+        args: {
+            modelVersionId: t.arg.string({ required: true }),
+            // TODO: Putting defaultValue & required overrides defaultValue
+            first: t.arg({
+                type: 'Limit',
+                defaultValue: 10,
+                required: true,
+            }),
+            after: t.arg.string(),
+            prefix: t.arg.string(),
+        },
+        async resolve(_root, args, _ctx) {
+            const splitKey = (prefixLength: number, objectKey?: string): string[] => {
+                if (objectKey) {
+                    return objectKey.slice(prefixLength + 1).split('/');
+                }
+
+                return [];
+            };
+
+            const folderOrFile = (keyArr?: string[]): 'folder' | 'file' | undefined => {
+                if (keyArr) {
+                    return keyArr.length > 1 ? 'folder' : 'file';
+                }
+
+                return undefined;
+            };
+
+            const modelStorageProvider = (await ObjectionMLModelVersion.relatedQuery(
+                'storageProvider',
+            )
+                .for(args.modelVersionId)
+                .first()) as ObjectionStorageProvider;
+
+            const modelVersion = (await ObjectionMLModelVersion.query()
+                .findById(args.modelVersionId)
+                .first()) as ObjectionMLModelVersion;
+
+            // Appends new folder name to base prefix
+            const prefix = `${modelVersion.s3Prefix}`.concat(args.prefix ? '/' + args.prefix : '');
+
+            /* If no continuation token, the s3 api will always
+               return the root directory as an object. We do not want
+               this returned to the client. */
+            const limit = !args.after ? args.first + 1 : args.first;
+
+            // To get Pothos and the S3 API to play nicely
+            const maxKeys =
```

Comment:
Dumb Q: would `maxKeys = args?.after` not do the same thing?

Reply:
Unfortunately no. `args.after` is of type `string | null | undefined`, and `maxKeys` must be of type `string | undefined`.

Reply:
Sorry, my brain dumb. `maxKeys = args?.after || undefined`

Just fussing with some sample null / undefined / string examples that feels like it should coerce nulls to undefined

![Screenshot_20240114-214159](https://github.com/dstk-labs/dstk/assets/1910269/b102dd9f-8c00-4e7b-ada0-4ea4f498fc8f)

Reply:
That'll do it!